src/firmware/app/control/trajectory_planner.c

```diff
@@ -0,0 +1,307 @@
+    // reversed again in the end)
+    if (t_end < t_start)
+    {
+        reverse_parameterization = true;
+
+        // Reverse the direction (Polynomial library can only handle forwards direction)
+        float temp = t_start;
+        t_start    = t_end;
+        t_end      = temp;
+
+        // Reverse the final and initial speeds
+        temp          = initial_speed;
+        initial_speed = final_speed;
+        final_speed   = temp;
+    }
+
+    // Create the parmeterization to contain the desired number of segments
+    CREATE_STATIC_ARC_LENGTH_PARAMETRIZATION(arc_length_param,
+                                             TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS);
+
+    // Get all of the points for the arc length parameterization (Not constant arc length
+    // segments)
+    shared_polynomial_getArcLengthParametrizationOrder3(path, t_start, t_end,
+                                                        arc_length_param);
+
+    const float arc_segment_length =
+        app_trajectory_planner_get_total_arcLength(arc_length_param) / num_segments;
+
+    // Now use numerical interpolation to get constant arc length segments for the
+    // parameterization
+    app_trajectory_planner_get_max_allowable_speed_profile(
+        max_allowable_speed_profile, traj_elements, path, num_segments, arc_length_param,
+        arc_segment_length, max_allowable_acceleration);
+
+    // First point on the velocity profile is always the current speed
+    velocity_profile[0] = initial_speed;
+
+    // Loop through the path forwards to ensure that the maximum velocity limit defined by
+    // the curvature is not breached by constant max acceleration of the robot (if it is,
+    // pull down the acceleration)
+    app_trajectory_planner_generate_forwards_continuous_velocity_profile(
+        num_segments, velocity_profile, max_allowable_speed_profile, arc_segment_length,
+        max_allowable_acceleration, max_allowable_speed);
+
+    // Now check backwards continuity. This is done to guarantee the robot can deccelerate
+    // in time to not breach the maximum allowable speed defined by the path curvature
+    velocity_profile[num_segments - 1] = final_speed;
+
+    app_trajectory_planner_generate_backwards_continuous_velocity_profile(
+        num_segments, velocity_profile, arc_segment_length, max_allowable_acceleration);
+
+    // Now that the velocity profile has been defined we can calculate the time profile of
+    // the trajectory
+    app_trajectory_planner_generate_time_profile(traj_elements, num_segments,
+                                                 arc_segment_length, velocity_profile);
+    //    traj_elements[0].time = 0.0;
+    //    for (unsigned int i = 0; i < num_segments - 1; i++)
+    //    {
+    //        traj_elements[i + 1].time =
+    //            traj_elements[i].time +
+    //            (2 * arc_segment_length) / (velocity_profile[i] + velocity_profile[i +
+    //            1]);
+    //    }
```

Comment:
We should remove this commented out code.

Reply:
Removed